pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -577,6 +571,13 @@ def is_subrepo(directory):
         with RpmPublication.create(
             repo_sync_results[PRIMARY_REPO], pass_through=False
         ) as publication:
+            gpgcheck = repository.repo_config.get("gpgcheck", 0)
+            has_repomd_signature = "repodata/repomd.xml.asc" in metadata_files_for_mirroring[repository].keys()
+
+            publication.package_checksum_type = CHECKSUM_TYPES.UNKNOWN
+            publication.metadata_checksum_type = CHECKSUM_TYPES.UNKNOWN
+            publication.repo_config = {"repo_gpgcheck": has_repomd_signature, "gpgcheck": gpgcheck}
+
             for path, repo_version in repo_sync_results.items():
                 add_metadata_to_publication(publication, repo_version, prefix=path)
```

Comment:
I also think we were re-setting these values over and over again for repositories with .treeinfo, and the end result values might have ended up being incorrect as a result.  We want to consider the parent repo only.